contributors/design-proposals/add-metadata-indicating-to-replace-union-in-patch.md

```diff
@@ -0,0 +1,415 @@
+Add Tag Indicating To Replace *Union* Fields in Patch (a.k.a Support Replacing Maps in Strategic Merge Patch)
+=============
+
+This approach could be used to fix the issues with `kubectl apply` simply by having apply always replace union fields instead of patching them.  E.g. instead of setting `spec.strategy.type` without touching the rest of `spec.strategy`, patches should always replace the entire `spec.strategy`.
+
+## Proposed Changes
+
+### APIs
+
+Add a go struct tag/openAPI value telling clients to replace the entire struct when doing a *PATCH*. For the *inlined* fields, we have 2 options:
```

Comment:
We should encourage discriminators:
https://github.com/kubernetes/kubernetes/issues/35345